### PR TITLE
FISH-6506: enable preprocessing for ENV placeholder on micro logging properties

### PR DESCRIPTION
**Pull request by breakponchito**
Enabling preprocessing for ENV placeholder on micro logging properties

## Description

This is a bug fix for an issue to map system properties to placeholders added on a logging.properties file for payara micro
### Testing Performed

Manual testing using the following command line for payara micro:
java -Dloglevel=FINE -jar payara-micro.jar --logProperties logging.properties

and using the attached logging.properties file attached on the ticket: [FISH-6506](https://payara.atlassian.net/browse/FISH-6506)

`~/projects/Payara/appserver/extras/payara-micro/payara-micro-distribution/target$ java -Dloglevel=FINE -jar payara-micro.jar --logProperties logging.properties

Sep 20, 2022 8:04:59 PM fish.payara.micro.impl.RuntimeDirectory processDirectoryInformation
WARNING: Payara Micro Runtime directory is located in a temporary file location which can be cleaned by system processes.

Sep 20, 2022 8:04:59 PM fish.payara.micro.impl.RuntimeDirectory processDirectoryInformation
INFO: Payara Micro Runtime directory is located at /tmp/payaramicro-rt5470083545123475589tmp

Sep 20, 2022 8:04:59 PM GlassFishRuntime getRuntimeBuilder
FINE: builder = com.sun.enterprise.glassfish.bootstrap.osgi.OSGiGlassFishRuntimeBuilder@1ee807c6

Sep 20, 2022 8:04:59 PM GlassFishRuntime getRuntimeBuilder
FINE: builder = com.sun.enterprise.glassfish.bootstrap.StaticGlassFishRuntimeBuilder@11c20519

Sep 20, 2022 8:04:59 PM GlassFishRuntime getRuntimeBuilder
FINE: builder = com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishRuntimeBuilder@365c30cc

Sep 20, 2022 8:04:59 PM GlassFishRuntime getRuntimeBuilder
FINE: builder = fish.payara.micro.boot.runtime.PayaraMicroRuntimeBuilder@5e25a92e

Sep 20, 2022 8:04:59 PM fish.payara.micro.boot.runtime.PayaraMicroRuntimeBuilder build
INFO: Built Payara Micro Runtime`

now the logs are not showing any error to set the ${ENV=loglevel} value

### Testing Environment

Ubuntu 20.04, Azul JDK 8, Maven 3.8.6

### CODE REVIEW

**Comment by breakponchito** [comment]
Jenkins test please

**Review by luiseufrasio — APPROVED** [review]
Tested locally in Community.
It works quite well!

**Comment by JamesHillyard** [comment]
Fixes https://github.com/payara/Payara/issues/5914

**Review by JamesHillyard — CHANGES_REQUESTED** [review]
This doesn't appear to work. 
- I setup an environment variable `loglevel` with value `FINE`
- I ran `java -jar payara-micro.jar --logproperties logging.properties`

No fine level logs could be seen in the output

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Yeah this doesn't fix it for me either

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
> This doesn't appear to work.
> 
> * I setup an environment variable `loglevel` with value `FINE`
> * I ran `java -jar payara-micro.jar --logproperties logging.properties`
> 
> No fine level logs could be seen in the output.
> 
> The command you are using contains `-Dloglevel=FINE` which shouldn't be required for this to work as expected.

I already fixed. The problem was that I didn't include environment variables, this was fixed by using the System.getenv() and I maintained the System.getProperties() to also include the variable from command line with the -D option if needed

**Review by JamesHillyard — COMMENTED** [review]
Tested and this seems to work well! Thank you

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 1]
```diff
@@ -1807,6 +1814,40 @@ private void resetLogging(String loggingProperty) {
...
+     * @throws IOException
+     */
+    private ByteArrayInputStream replaceEnvProperties(InputStream is) throws IOException {
+        //preprocessing the properties read from the custom properties file
+        Properties configuration = new Properties();
+        Properties environment = new Properties();
+        configuration.load(is);
+        //set the System.getProperties to be used for the replacement process. The desired property to be mapped
+        //should need to be available on the System properties
+        environment.putAll(System.getProperties());
```
Comment (Pandrex247) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:1830]: We're replacing ENV here - we shouldn't be substituting in system properties here.
System properties replace a different regex.

This reproducer is explicitly checking `${ENV=logLevel}`

Environment replacement = `${ENV=logLevel}`
System property replacement = `${logLevel}`
MicroProfile Config replacement = `${MPCONFIG=logLevel}`
Password Alias replacement = `${ALIAS=logLevel}`

https://docs.payara.fish/community/docs/Technical%20Documentation/Payara%20Server%20Documentation/Server%20Configuration%20And%20Management/Configuration%20Options/Variable%20Substitution/Types%20of%20Variables.html

```suggestion
```

Comment (Pandrex247) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:1830]: Whether we want to hold this PR back on us not performing the other substitutions is another question, to which my answer right now is "no, that's a separate larger issue which we can follow-up on"

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java (hunk 1 above)
Comment (breakponchito) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Ok based on the discussion I excluded the System.getProperties call and I just leave the System.getenv() to solve ENV variables replacement

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**